ibc_public/utils_labels.py

```diff
@@ -0,0 +1,83 @@
+import json
+import sys
+import pandas as pd
+
+import os
+
+_package_directory = os.path.dirname(os.path.abspath(__file__))
+
+ALL_CONTRASTS = os.path.join(_package_directory, '..', 'ibc_data',
+                             'all_contrasts.tsv')
+CONTRAST_COL = 'contrast'
+
+
+def get_labels(contrasts, contrast_col=CONTRAST_COL):
+    """
+    Returns the list of labels for each passed contrast name
+
+    Parameters
+    ----------
+
+    contrasts: list of str
+               each element of the list is a contrast name in the document
+
+    contrast_col: str
+                  string used to locate the column of the labels file that
+                  stores contrast names
+
+    Returns
+    -------
+
+    contrast_dict: dict
+                   dictionary containing the contrasts provided by the user
+                   as keys, and their corresponding labels as values
+    """
+    df = pd.read_csv(ALL_CONTRASTS, sep='\t')
+
+    contrast_dict = {}
+
+    for contrast in contrasts:
+        con = df[df[contrast_col] == contrast]
+
+        if len(con.index) == 0:
+            err = f"There is no contrast with the name {contrast}"
+            raise ValueError(err)
+        else:
+            labels = df.columns[con.isin([1.0]).any()]
+            contrast_dict[contrast] = [label for label in labels]
+
+    return contrast_dict
+
+
+def add_labels(contrast, labels, contrast_col=CONTRAST_COL,
+               output_file=ALL_CONTRASTS):
+    """
+    Adds all the passed labels to the selected contrast
+
+    Paramenters
+    -----------
+
+    contrast: str
+              name of the contrast that will get the labels
+
+    labels: list of str
+            labels that the user wants to add. The labels must exist as
+            columns in the file
+
+    contrast_col: str
+              string used to locate the column of the labels file that
+              stores contrast names
+
+    output_file: str or path object
+                 path to csv file where the new label database is to be saved
+                 with the changed
+    """
+    df = pd.read_csv(ALL_CONTRASTS, sep='\t')
+    con_index = df[df[contrast_col] == contrast].index
+    for label in labels:
+        if label in df.columns:
+            df.at[con_index, label] = 1.0
+        else:
+            print(f"There is no label with the name {label}")
```

Comment:
Why not add the label , then ?

Reply:
In case the user makes a typo and adds a useless label to the spreadshit, that could be cumbersome to deal with

Reply:
Sure, but this is why  we use git !
It is important that the structure moves on according to our needs. 
But indeed, we should at least issue a warning.

Reply:
Sure, we use git, but not everybody does. I think the clean way to do this is to create an add_label function, ask the user if they want to add the label, call this function and then set the value to 1. What do you think?

Reply:
As I told you yesterday, I now think that we should stop manipulating the table in its current formn, and only use a table with ('columns', 'cognitive labels') columns. Then it makes sense that somewody adding a new record can add the labels he/she wants.

Reply:
Done